sdonboarding.client/src/components/UITemplate/AddDataTemplate.jsx

```diff
@@ -0,0 +1,304 @@
+                const { customers, products, stores } = await getOtherObjectsData();
+                setLocalCustomers(customers);
+                setLocalProducts(products);
+                setLocalStores(stores);
+
+                //initialize localObject with proper values so that if there is no change triggered(onChange)
+                //we have the default first value to create the sale with
+                /*setLocalObject((prevState) => ({
+                    ...prevState,
+                    customerId: customers[0].id,
+                    customer: customers[0].name,
+                    productId: products[0].id,
+                    product: products[0].name,
+                    storeId: stores[0].id,
+                    store: stores[0].name,
+                }))*/
+            }
+            
+        } catch (error) {
+            console.error('Error fetching data:', error);
+            dispatch(setError('Error fetching data for Sales:', error));
+        }
+    }, [dispatch]);
+
+    useEffect(() => {
+       
+
+        fetchOthersData();
+    }, [fetchOthersData]);
+
+    const handleInputChange = (e) => {
+        const { name, value } = e.target;
+        e.target.setCustomValidity('');
+        setLocalObject((prevState) => ({
+            ...prevState,
+            [name]: value,
+        }));
+    };
+
+    /*handle the product active dropdown value*/
+    const handleActiveChange = (e) => {
+        setLocalObject((prevState) => ({
+            ...prevState,
+            active: e.target.value === 'true'
+        }));
+    };
+
+    /*List Change handler, it will update localObject with ID and name as we select it from the dropdown*/
+    const handleListChange = (id, name, type) => {
+        const selectedType = type;
+        if (selectedType === 'Customer') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                customerId: id,
+                customer: name,
+            }))
+        } else if (selectedType === 'Product') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                productId: id,
+                product: name,
+            }))
+        } else if (selectedType === 'Store') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                storeId: id,
+                store: name,
+            }))
+        }
+    };
+
+    const handleCancel = () => {
+        setLocalObject(null);
+        handleAddCancel();
+    };
+
+    
+    //handle the form submit event
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+
+        const valid = genericMethods.validateInputValuesOnSubmit(type);
+
+        //Call the parent function for edit request to server
+        if (valid === true) {
+            await handleAddSubmit();
+            setLocalObject(null);
+        } else {
+            console.log("Form is invalid");
+        }
```

Comment:
Don't include or expose any console.log in your code specially if is ready for deployment. It's fine with local but not on prod. You can also simplify your condition if you are handling boolean value. For example: **if (valid) {...}**